File: leetcode_python/medium/ContainerWithMostWater.py
'''
Created on Jan 11, 2015

@author: cksharma
'''
class Solution:
    # @return an integer
    def maxArea(self, height):
        i = 0; 
        j = len(height) - 1;
        ans = 0;
        while i < j:
            ans = max(ans, min(height[i], height[j]) * (j - i));
            if height[i] < height[j]:
                i += 1;
            else:
                j -= 1;
        return ans;
if __name__ == '__main__':
    sol = Solution();
    print (sol.maxArea([5, 4, 7, 8, 6, 3]));

File: leetcode/src/medium/ContainerWithMostWater.java
package medium;

public class ContainerWithMostWater {
	public int maxArea(int[] height) {
		int j = height.length - 1, i = 0, mx = 0;

		while (i < j) {
			mx = Math.max(mx, ((j - i) * (Math.min(height[i], height[j]))));

			if (height[i] < height[j])
				i++;
			else if (height[i] >= height[j])
				j--;
		}
		return mx;
	}

	public static void main(String[] args) {
		ContainerWithMostWater sol = new ContainerWithMostWater();
		System.out.println(sol.maxArea(new int[] { 5, 4, 7, 8, 6, 3 }));
	}
}
